cmake/checks/check_01_cxx_features.cmake

```diff
@@ -103,6 +103,10 @@ IF(NOT DEFINED DEAL_II_WITH_CXX14 OR DEAL_II_WITH_CXX14)
     MESSAGE(STATUS "Using C++ version flag \"${DEAL_II_CXX_VERSION_FLAG}\"")
     PUSH_CMAKE_REQUIRED("${DEAL_II_CXX_VERSION_FLAG}")
 
+    # Some versions of clang are feature complete but do not have debug
+    # information.
+    PUSH_CMAKE_REQUIRED("-g")
```

Comment:
Oh :-( I thought about having you approve the patch but then thought that we're so far behind in patch review that I should do my share in it and the patch looked reasonable to me. What is a better approach here?

By the way, shouldn't there be a POP_CMAKE_REQUIRED or similar as well?